StPicoEvent doesn't compile with modern compilers

On macOS with
```
clang version 7.1.0 (tags/RELEASE_710/final)
```

I get
```c++
./StPicoHelix.h:221:11: error: no member named 'finite' in the global namespace; did you mean 'isfinite'?
  if ( !::finite(mDipAngle) ) {
        ~~^~~~~~
          isfinite
/nix/store/znv06z6pld0ivwddj52rishggwc2m94c-libcxx-7.1.0-dev/include/c++/v1/math.h:439:1: note: 'isfinite' declared here
isfinite(_A1 __lcpp_x) _NOEXCEPT
^
In file included from StPicoPhysicalHelix.cxx:9:
./StPicoHelix.h:224:11: error: no member named 'finite' in the global namespace; did you mean 'isfinite'?
  if ( !::finite(mCurvature) ) {
        ~~^~~~~~
          isfinite
/nix/store/znv06z6pld0ivwddj52rishggwc2m94c-libcxx-7.1.0-dev/include/c++/v1/math.h:439:1: note: 'isfinite' declared here
isfinite(_A1 __lcpp_x) _NOEXCEPT
^
In file included from StPicoPhysicalHelix.cxx:9:
./StPicoHelix.h:243:13: error: no member named 'finite' in the global namespace; did you mean 'isfinite'?
    if ( !::finite( tmpVal ) ) {
          ~~^~~~~~
            isfinite
/nix/store/znv06z6pld0ivwddj52rishggwc2m94c-libcxx-7.1.0-dev/include/c++/v1/math.h:439:1: note: 'isfinite' declared here
isfinite(_A1 __lcpp_x) _NOEXCEPT
^
3 errors generated.
make: *** [Makefile:23: StPicoPhysicalHelix.o] Error 1
```

On macOS with
```
gcc version 10.3.0 (GCC)
```
```c++
In file included from StPicoPhysicalHelix.cxx:9:
StPicoHelix.h: In member function ‘Int_t StPicoHelix::bad(Double_t) const’:
StPicoHelix.h:221:11: error: ‘::finite’ has not been declared; did you mean ‘fwrite’?
  221 |   if ( !::finite(mDipAngle) ) {
      |           ^~~~~~
      |           fwrite
StPicoHelix.h:224:11: error: ‘::finite’ has not been declared; did you mean ‘fwrite’?
  224 |   if ( !::finite(mCurvature) ) {
      |           ^~~~~~
      |           fwrite
StPicoHelix.h:243:13: error: ‘::finite’ has not been declared; did you mean ‘fwrite’?
  243 |     if ( !::finite( tmpVal ) ) {
      |             ^~~~~~
      |             fwrite
```

## Steps to reproduce ##
```
cd StRoot/StPicoEvent
make
```

Also (again with clang):
```
StPicoHelix.cxx:634:13: error: invalid operands to binary expression ('std::ostream' (aka 'basic_ostream<char>') and 'char')
  return os << '('
         ~~ ^  ~~~
```

```
StPicoEvent.cxx:332:9: warning: implicit conversion from 'std::__1::numeric_limits<unsigned short>::type' (aka 'unsigned short') to 'UChar_t'
      (aka 'unsigned char') changes value from 65535 to 255 [-Wconstant-conversion]
                      std::numeric_limits<unsigned short>::max() :
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/StRoot/StPicoEvent/StPicoEvent.cxx
+++ b/StRoot/StPicoEvent/StPicoEvent.cxx
@@ -328,8 +328,8 @@ void StPicoEvent::setBunchId(Int_t id) {
     LOG_INFO << "StPicoEvent::setBunchID() - negative bunch ID = " << id << endm;
   }
   else {
-    mBunchCrossId = ( ( id > std::numeric_limits<unsigned short>::max() ) ?
-		      std::numeric_limits<unsigned short>::max() :
+    mBunchCrossId = ( ( id > std::numeric_limits<decltype(mBunchCrossId)>::max() ) ?
+		      std::numeric_limits<decltype(mBunchCrossId)>::max() :
 		      (UChar_t)id );
   }
 }

--- a/StRoot/StPicoEvent/StPicoHelix.cxx
+++ b/StRoot/StPicoEvent/StPicoHelix.cxx
@@ -14,6 +14,7 @@
 
 // C++ headers
 #include <float.h>
+#include <ostream>
 
 // PicoDst headers
 #include "StPicoHelix.h"

--- a/StRoot/StPicoEvent/StPicoHelix.h
+++ b/StRoot/StPicoEvent/StPicoHelix.h
@@ -13,7 +13,7 @@
 #define StPicoHelix_h
 
 // C++ headers
-#include <math.h>
+#include <cmath>
 #include <utility>
 #include <algorithm>
 
@@ -218,10 +218,10 @@ inline Double_t StPicoHelix::pathLength(Double_t X, Double_t Y) const { return f
 inline Int_t StPicoHelix::bad(Double_t WorldSize) const {
 
   Int_t ierr;
-  if ( !::finite(mDipAngle) ) {
+  if ( !std::isfinite(mDipAngle) ) {
     return 11;
   }
-  if ( !::finite(mCurvature) ) {
+  if ( !std::isfinite(mCurvature) ) {
     return 12;
   }
   
@@ -240,7 +240,7 @@ inline Int_t StPicoHelix::bad(Double_t WorldSize) const {
     default: tmpVal = NAN;
     };
     
-    if ( !::finite( tmpVal ) ) {
+    if ( !std::isfinite( tmpVal ) ) {
       ierr = 10 + iIter;
     }
     if ( ::fabs( tmpVal ) > WorldSize ) {

--- a/StRoot/StPicoEvent/StPicoHelix.h
+++ b/StRoot/StPicoEvent/StPicoHelix.h
@@ -217,7 +217,7 @@ inline Double_t StPicoHelix::pathLength(Double_t X, Double_t Y) const { return f
 
 inline Int_t StPicoHelix::bad(Double_t WorldSize) const {
 
-  Int_t ierr;
+  Int_t ierr = 0;
   if ( !std::isfinite(mDipAngle) ) {
     return 11;
   }

